[Bug] compile hyperscan failed when build thirdparty

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.


### Version

master

### What's Wrong?


```
FAILED: src/parser/Parser.cpp /home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/src/hyperscan-5.4.0/doris_build/src/parser/Parser.cpp
cd /home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/src/hyperscan-5.4.0/doris_build && /home/disk4/xuyang/work/baidu/bdg/doris/palo-toolchain/ldb_toolchain/bin/cmake -E make_directory /home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/src/hyperscan-5.4.0/doris_build/src/parser && /home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/installed/bin/ragel /home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/src/hyperscan-5.4.0/src/parser/Parser.rl -o /home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/src/hyperscan-5.4.0/doris_build/src/parser/Parser.cpp
/home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/installed/bin/ragel: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by /home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/installed/bin/ragel)
/home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/installed/bin/ragel: /usr/lib64/libstdc++.so.6: version `CXXABI_1.3.8' not found (required by /home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/installed/bin/ragel)
/home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/installed/bin/ragel: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/installed/bin/ragel)
/home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/installed/bin/ragel: /usr/lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/installed/bin/ragel)
/home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/installed/bin/ragel: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /home/disk4/xuyang/work/baidu/bdg/doris/core/thirdparty/installed/bin/ragel)
[135/344] Building C object CMakeFiles/hs_exec_avx2.dir/src/rose/program_runtime.c.o
ninja: build stopped: subcommand failed.
```


### What You Expected?

work

### How to Reproduce?

Clean the thirdparty.
Build be.
And compile error occur.

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/thirdparty/build-thirdparty.sh
+++ b/thirdparty/build-thirdparty.sh
@@ -605,6 +605,13 @@ build_hyperscan() {
     check_if_source_exist "${RAGEL_SOURCE}"
     cd "${TP_SOURCE_DIR}/${RAGEL_SOURCE}"
 
+    if [[ "${KERNEL}" != 'Darwin' ]]; then
+        cxxflags='-static'
+    else
+        cxxflags=''
+    fi
+
+    CXXFLAGS="${cxxflags}" \
     ./configure --prefix="${TP_INSTALL_DIR}"
     make install
 

